### PR TITLE
Finish round button

### LINKED ISSUE #1826: Finish round button
Instead of ending the round automatically when the last jurisdiction finishes auditing, let the audit admin press a button to end the round.

There should be some guards to make sure they don't end the round before reviewing discrepancies.

### PR DESCRIPTION
**Pull request by jonahkagan**
Closes #1826, https://github.com/votingworks/arlo/issues/1541

_Review by commits_

Instead of ending the round automatically when the last jurisdiction finishes auditing, lets the audit admin press a button to end the round.

https://github.com/votingworks/arlo/assets/530106/4dd146b1-7d5d-47b7-9be3-b8b976f79839

### CODE REVIEW

File: server/api/rounds.py [hunk 1]
```diff
@@ -519,6 +616,31 @@ def create_round(election: Election):
     return jsonify({"status": "ok"})
 
 
+@api.route("/election/<election_id>/round/<round_id>/finish", methods=["POST"])
```
Comment (arsalansufi) [inline server/api/rounds.py:619]: Is it ever the case that we need to finish a round other than the most current? Wondering if this endpoint really requires a `round_id` param

**Review by arsalansufi — APPROVED** [review]
One q, otherwise LGTM!

File: server/api/rounds.py (hunk 1 above)
Comment (jonahkagan) [reply server/api/rounds.py:619]: Good q. No, we don't need to finish other rounds.

The convention in Arlo has been to overspecify params in the API urls and then verify them in the endpoint (e.g. check out the pattern in `undo_round_start`). So this endpoint should validate that the given round id is the current round (which I'm now realizing is missing here).

I'm feeling unsure whether it's better to change to the style you suggest (which I agree is simpler), or to stick to the more REST-y convention of specifying explicitly the resource you want to modify.

Comment (arsalansufi) [reply server/api/rounds.py:619]: Got it - I could go either way on this one. I do like consistency, but also think that it's okay to diverge from this convention if it isn't doing much for us

Comment (jonahkagan) [reply server/api/rounds.py]: Decided to update to `/election/<election_id>/rounds/current/finish` to still have a bit of that REST-y vibe will still getting the benefits of the simplification. And updated the undo round start endpoint to match. Thanks for the suggestion!

Comment (arsalansufi) [reply server/api/rounds.py]: Nice, I like that compromise!

**Pull request merged**